### PR TITLE
ci: remove k8s 1.30 tests check

### LINKED ISSUE #4625: ci: k8s 1.30 tests are excluded for release-v3.11
k8s 1.30 tests are excluded for release-v3.11 here
https://github.com/ceph/ceph-csi/blob/4c7d419336b6c65083609010d250ca634db391c7/.github/workflows/pull-request-commentor.yaml#L23-L32

but mergify has status check condition for k8s 1.30 tests
https://github.com/ceph/ceph-csi/blob/4c7d419336b6c65083609010d250ca634db391c7/.mergify.yml#L29-L49

need to be fixed in separate PR.  

for now will trigger manually k8s 1.30 tests, hopefully it should go through🤞

cc @Madhu-1 @Rakshith-R

### PR DESCRIPTION
**Pull request by iPraveenParihar**
# Describe what this PR does #

k8s 1.30 tests were excluded for release-v3.11 but
mergify has status check condition for k8s 1.30 tests.
This commit removes these checks from mergify.

## Related issues ##

Closes: #4625 

**Checklist:**

* [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#commit-messages).
* [x] Reviewed the developer guide on [Submitting a Pull Request](https://github.com/ceph/ceph-csi/blob/devel/docs/development-guide.md#development-workflow)
* [x] [Pending release notes](https://github.com/ceph/ceph-csi/blob/devel/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next major release.
* [x] Documentation has been updated, if necessary.
* [x] Unit tests have been added, if necessary.
* [x] Integration tests have been added, if necessary.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

* `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)

</details>

### CODE REVIEW

File: .mergify.yml [hunk 1]
```diff
@@ -37,15 +37,12 @@ queue_rules:
               - "status-success=ci/centos/k8s-e2e-external-storage/1.27"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.28"
               - "status-success=ci/centos/k8s-e2e-external-storage/1.29"
-              - "status-success=ci/centos/k8s-e2e-external-storage/1.30"
```
Comment (Rakshith-R) [inline .mergify.yml:40]: There's one more set here https://github.com/ceph/ceph-csi/blob/c96e05a4c257ebafbf587a9f7f0ac0f991585cb4/.mergify.yml#L268-L294

Comment (Madhu-1) [reply .mergify.yml:40]: instead of this one i would like to run 1.30 test as we might need to do minor releases due to bug fixes.

Comment (iPraveenParihar) [reply .mergify.yml:40]: okay, then I'll include the 1.30 tests for `release-v3.11`

File: .github/workflows/pull-request-commentor.yaml [hunk 2]
```diff
@@ -27,10 +27,6 @@ jobs:
           - k8s: "1.26"
             branch: "devel"
 
-          # Ceph-CSI <= 3.11 was released before Kubernetes 1.30
-          - k8s: "1.30"
-            branch: "release-v3.11"
-
```
Comment (Madhu-1) [inline .github/workflows/pull-request-commentor.yaml]: Sorry for a mistake from my side, now i see that we don't support 1.30 with even patch releases. can you please remove 1.30 check from mergify.yaml where we have `base~=^(release-.+)$` check and nothing else.

**Review by Madhu-1 — APPROVED** [review]

**Review by Rakshith-R — APPROVED** [review]
Thanks

**Comment by iPraveenParihar** [comment]
We also need to backport this to v3.11

**Comment by Madhu-1** [comment]
@iPraveenParihar mergify rules are checked on only main/devel branch, if required we can backport it later.

**Comment by Madhu-1** [comment]
@mergifyio queue

**Pull request merged**